wlroots/xwayland.py

```diff
@@ -0,0 +1,462 @@
+# Copyright (c) 2022 Matt Colligan
+
+from __future__ import annotations
+
+import enum
+from typing import TYPE_CHECKING
+
+from pywayland.server import Signal
+
+from wlroots import ffi, lib, Ptr, PtrHasData, str_or_none
+from wlroots.wlr_types.surface import Surface as WlrSurface
+
+if TYPE_CHECKING:
+    from pywayland.server import Display
+
+    from wlroots.wlr_types import Compositor, Seat
+    from wlroots.wlr_types.xdg_shell import SurfaceCallback, T
```

Comment:
The `T` here is just a `TypeVar`, I would suggest creating a new `T` here instead of importing this one.